### DISCUSSION TITLE
离线安装第三方库的源码的问题

### DISCUSSION THREAD

**Discussion by zyjdmmm** [xmake-io/xmake-repo#2149] [Q&A]
### 你在什么场景下需要该功能？

需求:
希望能在工程目录下比如说/third_part下保存一份第三方库的源码，以便于网络被墙的时候换个没有梯子的电脑也能编译，还有就是公司有的电脑不能联网。。。

### 描述可能的解决方案

1. 首先我在工程目录的/third_part下放置源码压缩包，但是我发现使用'xmake g --pkg_searchdirs="/third_part"'后xmake依然要联网下载源码（我换了个压缩格式，难道是包名字不对？）

![捕获](https://user-images.githubusercontent.com/75295100/228566808-0fca33d9-e348-48af-9470-ab5e925ad0d3.PNG)
![捕获1](https://user-images.githubusercontent.com/75295100/228566817-f53adcd1-cdfe-449e-9881-c2329da6d8db.PNG)

2. 希望能有更优雅一点的方式保存源码，比如直接在/third_part下放置解压缩后的第三方库源码也行。目的就是说换台不联网的电脑也能一键编译。
3. 'xmake require --export pcre'这个似乎只能导出编译好的包，有没有类似于'xmake require --export all'导出所有包（当然我知道可以去.xmake下直接拷贝）

### 描述你认为的候选方案

直接在/third_part下放置解压缩后的第三方库源码

**Comment by waruqi** [comment]
> 但是我发现使用'xmake g --pkg_searchdirs="/third_part"'后xmake依然要联网下载源码

win 下有 `/third_part` 这种绝对根路径？

另外搜索名通过 `xrepo info ` 看，文档有详细描述，仔细看下

```
$ xrepo info zlib
The package info of project:
..
      -> searchnames:
         -> zlib-1.2.13.tar.gz
         -> zlib-v1.2.13.tar.gz
         -> v1.2.13.tar.gz
         -> zlib (git)

```
